### DISCUSSION TITLE
Add main overview for the new comers

### DISCUSSION THREAD

**Discussion by einenlum** [phpactor/phpactor#2248] [General]
For now it is a bit complicated for new comers to get into the project, because we don't have the big picture (explain the workflow with extensions, the decoupling of the library, how the RPC can work…). And then maybe explain the next steps, the roadmap.

**Comment by mamazu** [comment]
I will make this my next project and continue on documenting how the diagnostic providers work and how the system works in general.

**Comment by dantleech** [comment]
note the language server actually has documentation here: https://language-server-platform.readthedocs.io/en/latest/reference/service-providers.html

also note that things can still change, and some things may be worth documenting while others not, so probably worth statring a discussion before comitting to anything.
